core/domain/question_domain.py

```diff
@@ -1503,6 +1503,8 @@ def partial_validate(self) -> None:
             if answer_group.outcome.dest_if_really_stuck is not None:
                 dest_if_stuck_is_specified = True
 
+        # Ruling out the possibility of None for mypy type checking.
```

Comment:
Ditto, needs justification.

Reply:
Whenever we create a default question, we create a `default_question_state_data` and while creating a `default_question_state_data` we are providing an outcome to that default question.

So, `create_default_question` will call `create_default_question_state`.
https://github.com/oppia/oppia/blob/a65d963b3268c484a2fafa6778979a79c5effefe/core/domain/question_domain.py#L1592
 which calls `create_default_state`
https://github.com/oppia/oppia/blob/a65d963b3268c484a2fafa6778979a79c5effefe/core/domain/question_domain.py#L258
 `create_default_state` calls -> `create_default_interaction`
https://github.com/oppia/oppia/blob/933bfccb33903dff2025a1590126326e5c7e4fca/core/domain/state_domain.py#L3855

and here we create a default outcome:
https://github.com/oppia/oppia/blob/933bfccb33903dff2025a1590126326e5c7e4fca/core/domain/state_domain.py#L976